### PR TITLE
#1618: Duplicated items in Used in Entities filters

### LINKED ISSUE #1618: Duplicated items in Used in Entities filters
### Description
The issue is not reproducible constantly, but I have seen it on all **Used in** filters.
Usually, after the web page is refreshed, the issue disappears 
I will continue to investigate it and will add additional info if I find any
A low Severity is added

### Precondition (*)
1. Install Magento with Adobe Stock Integration
2. Enhanced Media Gallery is Enabled
3. A saved image Preview from Adobe Stock

### Steps to reproduce (*)
1. Go to **Content - Media Gallery**
2. Click **Filters**
3. Expand, for example, **Used in Pages** filter
4. Start typing _page_ into **Page Title** field

### Actual result (*)
The found entities are duplicated
duplicated
duplicated2

### PR DESCRIPTION
**Pull request by joweecaquicla**
### Description (*)

This PR will introduce the fix to the issue wherein duplicate items appear in Used in Entities filters.
### Fixed Issues (if relevant)

1. Fixes magento/adobe-stock-integration https://github.com/magento/adobe-stock-integration/issues/1618: Duplicated items in Used in Entities filters
2. ...

### Manual testing scenarios (*)

1. Go to **Content - Media Gallery**
2. Click **Filters**
3. Expand, for example, Used in Pages filter
4. Start typing page into Page Title field
5. There should be no duplicates in the options.

### CODE REVIEW

**Comment by joweecaquicla** [comment]
@magento run all tests

**Comment by joweecaquicla** [comment]
@magento run all tests

File: MediaGalleryCmsUi/view/adminhtml/web/js/form/element/used-in-ui-select.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+], function (Select, $, _) {
+    'use strict';
+
+    return Select.extend({
+
+        /**
+         * @inheritdoc
+         */
+        success: function (response) {
+            var options = [];
```
Comment (sivaschenko) [inline MediaGalleryCmsUi/view/adminhtml/web/js/form/element/used-in-ui-select.js:19]: Options reset at each request success will most likely break the pagination functionality:
 - Generate more than 50 products (`bin/magento setup:perf:generate-fixtures setup/performance-toolkit/profiles/ce/small.xml`
 - Go to media gallery and type "Product" to "Used in Products" filter search field
 - Scroll to the bottom of the loaded options in the select
 - Next 50 products should be loaded

(it's easier to reproduce with products as we have the fixture to generate them, but the same behaviour should be preserved for all entities)

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @joweecaquicla ! Please see my review comment

**Pull request closed without merging**